x/mongo/driver/topology/topology.go

```diff
@@ -673,3 +673,41 @@ func (t *Topology) String() string {
 	}
 	return fmt.Sprintf("Type: %s, Servers: [%s]", desc.Kind, serversStr)
 }
+
+// isTopologyConsistent returns false when we have a replica set that claims to
+// have no primary but there exists a primary with all other nodes as secondaries. This
+// specifically works around HELP-13825.
+func (t *Topology) IsConsistent() bool {
+	desc := t.Description()
+
+	if desc.Kind != description.ReplicaSetNoPrimary {
+		return true
+	}
+
+	t.serversLock.Lock()
+	serversCopy := make([]*Server, 0, len(t.servers))
+	for _, s := range t.servers {
+		serversCopy = append(serversCopy, s)
+	}
+	t.serversLock.Unlock()
+
+	if len(serversCopy) == 0 || len(serversCopy) == 1 {
```

Comment:
q: isn't it still "inconsistent" if `len(serversCopy) == 1` and the one server listed is in the "secondary" state?
or i guess to put it a different way, is there any specific reason why the criteria applied in the loop below can't also apply to a one-node topology?

Reply:
that's true, but is it possible for a one-node topology to be listed as a secondary?

Reply:
it's very unlikely, but the node could also be in any other non-primary state (startup, recovering, down, etc.) - not just "secondary"

Reply:
interesting, that makes sense. If the only node was in another state, I would expect the client to fail on connections anyhow - regardless of if we tried to reconnect using this consistency check.